### ISSUE TITLE
Definition of the word/backwards motions (vi mode) are incorrect

### ISSUE THREAD

**Issue by Stebalien** [fish-shell/fish-shell#3285]
**fish version installed** _(`fish --version`)_: 2.3.1

**OS/terminal used**: Arch Linux

The definition of the word motion (`w`) for in the Vi key bindings is incorrect.
## Reproduction steps
1. Launch fish
2. run fish_vi_keybindings
3. Type `a./b/c`
4. Hit ESC.
5. Go to the beginning of the line (`^`).
6. Hit `w` three times.
## Expected results

The cursor lands on `.`, then `b`, then the last `/`. 
## Actual results

The cursor lands on the first `/`, then `c`, then _after_ the end of the line.

---

The `b` (word backwards) motion is similarly incorrect (try walking backwards over the same path in both vi and fish).

**Comment by krader1961** [comment]
Yes, fish's vi mode emulation differs from an actual Vi editor in numerous ways. Patches to improve the behavior are welcomed :smile:

**Comment by dpacbach** [comment]
Would like to point out that this incorrect behavior extends to underscores (`_`) as well.  In VI, the underscore is not considered a word boundary, but it is in fish, which I think is definitely the wrong behavior.  For example:
```
fish> this_is_a_test
```
When hitting `w`, the cursor stops on the underscores, when it should not.

**Comment by jdanbrown** [comment]
+1 for excluding underscores from word boundaries. This was one of the very first annoyances I encountered when attempting to migrate from bash with vi line-editing mode.
```
fish> this_is_a_test
```

**Comment by krobelus** [comment]
fixed by bbb2f0de8d5 (feat(vi-mode): make word movements vi-compliant, 2026-01-10)
